frame/system/src/limits.rs

```diff
@@ -342,13 +350,16 @@ impl BlockWeights {
 		BlockWeightsBuilder {
 			weights: BlockWeights {
 				base_block: constants::BlockExecutionWeight::get(),
-				max_block: Weight::zero(),
+				max_block: Weight::zero(), // This will be set by `build`.
 				per_class: PerDispatchClass::new(|class| {
-					let initial =
-						if class == DispatchClass::Mandatory { None } else { Some(Weight::zero()) };
+					let initial = if class == DispatchClass::Mandatory {
+						WeightLimit::UNLIMITED
+					} else {
```

Comment:
Here the `Mandatory` gets its special role. By having this, we immediately pull every blocks' `max_size` up to `(u64::MAX, u64::MAX)` since this invalidates any effective maximal block size.